tests/unit/test_utils.py

```diff
@@ -1837,7 +1837,7 @@ def test_flatten_dict_no_nested(self):
 
     def test_flatten_dict_with_various_types(self):
         nested_dict = {"a": [1, 2, 3], "b": {"x": None, "y": {"p": [], "q": ""}}, "c": 9}
-        flattened_dict = {"a": [1, 2, 3], "b.x": None, "b.y.p": [], "b.y.q": "", "c": 9}
+        flattened_dict = {"a": [1, 2, 3], "b^x": None, "b^y^p": [], "b^y^q": "", "c": 9}
```

Comment:
Can you add a unit test that combines instance-specific metadata, to re-create the situation Kyle had tested?

Reply:
I'll address this in my next PR which the tools will be replaced